types/signed_header.go

```diff
@@ -22,7 +25,49 @@ func (sH *SignedHeader) Verify(untrst header.Header) error {
 		// and is a programmer bug
 		panic(fmt.Errorf("%T is not of type %T", untrst, untrstH))
 	}
+	err := untrstH.ValidateBasic()
+	if err != nil {
+		return &header.VerifyError{
+			Reason: err,
+		}
+	}
+	sHHash := sH.Header.Hash()
+	if !bytes.Equal(untrstH.LastHeaderHash[:], sHHash) {
+		return &header.VerifyError{
+			Reason: fmt.Errorf("Last header hash %v does not match hash of previous header %v", untrstH.LastHeaderHash[:], sHHash),
+		}
+	}
+	sHLastCommitHash := sH.getLastCommitHash()
+	if !bytes.Equal(untrstH.LastCommitHash[:], sHLastCommitHash) {
+		return &header.VerifyError{
+			Reason: fmt.Errorf("Last commit hash %v does not match hash of previous header %v", untrstH.LastCommitHash[:], sHHash),
+		}
+	}
 	return sH.Header.Verify(&untrstH.Header)
 }
 
 var _ header.Header = &SignedHeader{}
+
+func (sH *SignedHeader) getLastCommitHash() []byte {
+	lastABCICommit := tmtypes.Commit{
+		Height: sH.Height(),
+		Round:  0,
+		BlockID: tmtypes.BlockID{
+			Hash:          tmbytes.HexBytes(sH.Hash()),
+			PartSetHeader: tmtypes.PartSetHeader{},
+		},
+	}
+	for _, sig := range sH.Commit.Signatures {
+		commitSig := tmtypes.CommitSig{
+			BlockIDFlag: tmtypes.BlockIDFlagCommit,
+			Signature:   sig,
+		}
+		lastABCICommit.Signatures = append(lastABCICommit.Signatures, commitSig)
+	}
+
+	if len(sH.Commit.Signatures) == 1 {
```

Comment:
can we add a comment to this? It's not obvious what we are doing here and why it only applies to len of 1